File: docs/src/testers/functions/invocations/timeout_examples.py
# @sniptest filename=timeout_examples.py
# @sniptest show=6-15
# ruff: noqa: E402 -- hidden live fixture precedes displayed imports
import os

function_id = os.environ["NOTTE_FUNCTION_ID"]

from notte_sdk import NotteClient

client = NotteClient()
function = client.Function(function_id=function_id)

# Short task
result = function.run(url="https://example.com", timeout=60)

# Long task
result = function.run(url="https://example.com", timeout=600)

assert result.status == "closed"
assert result.result == {"url": "https://example.com", "search_query": ""}

File: docs/src/testers/functions/invocations/timeout_examples.ts
// @sniptest filename=timeout_examples.ts
// @sniptest show=5-14
import assert from 'node:assert/strict';

const functionId = process.env.NOTTE_FUNCTION_ID!;

import { NotteClient } from 'notte-sdk';

const client = new NotteClient();
const fn = client.NotteFunction({ function_id: functionId });

// Short task
let result = await fn.run({ url: 'https://example.com' }, { timeoutMs: 60_000 });

// Long task
result = await fn.run({ url: 'https://example.com' }, { timeoutMs: 600_000 });

assert.equal(result.status, 'closed');
assert.deepEqual(result.result, { url: 'https://example.com', search_query: '' });
export { result };
